lib/endpoints/vms.js

```diff
@@ -213,6 +218,18 @@ function listVms(req, res, next) {
 
             res.header('x-joyent-resource-count', count);
 
+            var limit = DEFAULT_LIST_VM_LIMIT;
+            if (req.params.limit)
+                limit = Number(req.params.limit);
+
+            params.limit = Math.min(limit, MAX_LIST_VMS_LIMIT);
```

Comment:
Yes. that's definitely a better approach. Will fix soon before this PR is merged.